### ISSUE TITLE
Mail index empty with imap_check_subscribed

### ISSUE THREAD

**Issue by hasufell** [neomutt/neomutt#2312]
## Expected Behaviour

Emails showing up at the index page.

## Actual Behaviour

No emails shown (although inbox header cache is visibly loaded). Have to re-open the INBOX after start.

Screenshot shows I am inside the INBOX:

![mutt_bug](https://user-images.githubusercontent.com/1241845/80927469-6eb61f80-8d9e-11ea-981c-6d2b46a3363c.png)

Also note: There seem to be random "Mailbox closed" messages when both starting and switchting to other folders. Googling this it seems many mutt users have this.

## Steps to Reproduce

* Use the following config with an imap server:
https://gist.github.com/hasufell/fe227d5dfb0b5f2de63b98f190abc6a7#file-gistfile1-txt
* start neomutt

## How often does this happen?

- Always

## NeoMutt Version

https://gist.github.com/hasufell/fe227d5dfb0b5f2de63b98f190abc6a7#file-gistfile2-txt

## Extra Info

* OS: Exherbo
* No tmux

**Comment by zsugabubus** [comment]
Empty lines probably because #2307. 😄  “Mailbox closed” just in our cases, but hopefully no garbage text shown now.

People more clever than me also can explain why “mailbox closed”, but probably they will require a log file too.

**Comment by gahr** [comment]
My guess is the IMAP server is sending back an error response for some reason, and that confuses NeoMutt. I'm trying to fix that in #2303.

Please run with -d4 and post the log file so we can figure why that is happening.

**Comment by hasufell** [comment]
> Unsubscribing from that should solve the problem for you.

I have no idea how to do that and googling this doesn't yield much useful answers, except people pointing out how the IMAP protocol works.

> Please run with -d4 and post the log file so we can figure why that is happening.

The log file contains personal data. Can you tell me what to look for rather?

**Comment by gahr** [comment]
From NeoMutt, you can use `:unsubscribe-from Foo` where `Foo` is the missing mailbox you're still subscribed to. First we need to find which one it is.

I am looking for a `NO` response from the server after NeoMutt issues an `LSUB`. I'm interested in everything between a line such as
`[2020-05-04 09:21:49]<2> mutt_socket_write_d() 5> a0005 LSUB "" "*"`
and a line such as 
`   276 [2020-05-04 09:21:49]<2> mutt_socket_write_d() 5> a0007 FETCH 1:11067 (UID FLAGS INTERNALDATE RFC822.SIZE BODY.PEEK[HEADER.FIELDS (DATE FROM SENDER SUBJECT TO`

You might not see the second line if something goes wrong handling `LSUB`.

**Comment by hasufell** [comment]
```
[2020-05-04 10:52:02]<2> mutt_socket_write_d() 5> a0004 LSUB "" "*"
a0005 SELECT "INBOX"
[2020-05-04 10:52:02]<3> cmd_parse_lsub() Subscribing to ...
[2020-05-04 10:52:02]<3> imap_mdata_new() hcache uidvalidity 1376512568, uidnext 411, modseq 0
[2020-05-04 10:52:02]<3> cmd_parse_lsub() Subscribing to ...
[2020-05-04 10:52:02]<3> imap_mdata_new() hcache uidvalidity 1376512562, uidnext 18353, modseq 0
[2020-05-04 10:52:02]<3> cmd_parse_lsub() Subscribing to ...
[2020-05-04 10:52:02]<3> imap_mdata_new() hcache uidvalidity 1443785545, uidnext 3155, modseq 0
[2020-05-04 10:52:02]<3> cmd_parse_lsub() Subscribing to ...
[2020-05-04 10:52:02]<3> cmd_parse_lsub() Subscribing to ...
............................................................
[2020-05-04 10:52:02]<3> imap_mbox_open() Getting mailbox FLAGS
[2020-05-04 10:52:02]<3> imap_mbox_open() Getting mailbox PERMANENTFLAGS
[2020-05-04 10:52:02]<2> cmd_parse_exists() Handling EXISTS
[2020-05-04 10:52:02]<2> cmd_parse_exists() New mail in INBOX - 8456 messages total
[2020-05-04 10:52:02]<3> imap_mbox_open() Getting mailbox UIDVALIDITY
[2020-05-04 10:52:02]<3> imap_mbox_open() Getting mailbox UIDNEXT
[2020-05-04 10:52:02]<3> imap_mbox_open() Getting mailbox HIGHESTMODSEQ
[2020-05-04 10:52:02]<3> imap_cmd_step() IMAP queue drained
[2020-05-04 10:52:02]<3> imap_mbox_open() Mailbox flags: [\Answered] [\Flagged] [\Deleted] [\Seen] [\Draft] [$label1] [NonJunk] [Junk] [$Junk] [$NotJunk] [$label3] [$Forwarded] [Old] [\*] 
```

**Comment by gahr** [comment]
Nothing wrong in there - can you please post the whole log file? Just remove whatever looks sensitive.

**Comment by hasufell** [comment]
> Nothing wrong in there - can you please post the whole log file? Just remove whatever looks sensitive.

Those are **18.000** lines of log file. Since I've already spotted my password being plain base64 encoded in the log, I don't know what else is in there.

**Comment by hasufell** [comment]
Also note: I just tried with `mutt` and it doesn't have that issue.

**Comment by gahr** [comment]
Thanks - you'll have to help me help you.

The log file might contain indications that the IMAP server errored out at some point. I don't know which point so I can't tell you which line number to look at. All I can do is ask you to please post the log file, after making sure you remove sensitive information. I'm sorry I don't have a more pleasant way.

Or we can do trial and error.

* NeoMutt parses 8456 messages. Can you confirm that the number of emails in INBOX is correct?
* Can you try to run with an empty header cache?
* Can you try to open any other mailbox on startup and see whether the problem goes away?
* Can you use any other client (like a web mailer) and see whether you are subscribed to any mailboxes that doesn't exist?

We can try that from the command line:

1. produce your auth string by combining your username and password and base64 encoding them in the string "\0user\0pass". On my system, that'd be `printf "\0user\0pass" | b64encode -r -`.
2. connect to your IMAP server. On my system, that'd be `openssl s_client -host 192.168.1.1 -port 993`.
3. use the string produced at 1. to authenticate by issuing `a1 authenticate plain <the string>`
4. list the subscribed folders by issuing `a2 lsub "" "*"`
5. check whether any mailbox in that list doesn't exist; if so, unsubscribe from them using `a3 unsubscribe <mailbox-name>`

Let's see where we get from here.

**Comment by hasufell** [comment]
Yes, I found a mailbox that didn't exist anymore and now the issue seems to be gone. So this is a bug in neomutt.

**Comment by gahr** [comment]
Nobody's saying otherwise. As per my previous comment:

> My guess is the IMAP server is sending back an error response for some reason, and that confuses NeoMutt. I'm trying to fix that in #2303.

Thanks for confirming that.
